packages/@o3r/training-sdk/project.json

```diff
@@ -0,0 +1,52 @@
+{
+  "name": "training-sdk",
+  "$schema": "https://raw.githubusercontent.com/nrwl/nx/master/packages/nx/schemas/project-schema.json",
+  "projectType": "library",
+  "sourceRoot": "packages/@o3r/training-sdk/src",
+  "prefix": "o3r",
+  "targets": {
+    "build": {
+      "executor": "nx:noop",
+      "dependsOn": ["compile", "extract-folder-structure"]
+    },
+    "compile": {
+      "executor": "nx:run-script",
+      "options": {
+        "script": "build"
+      },
+      "inputs": ["source", "^source-and-cli"]
+    },
+    "regen": {
+      "executor": "nx:run-script",
+      "options": {
+        "script": "spec:regen"
+      }
+    },
+    "lint": {
+      "options": {
+        "eslintConfig": "packages/@o3r/training-sdk/.eslintrc.js",
+        "lintFilePatterns": [
+          "packages/@o3r/training-sdk/src/**/*.ts",
+          "packages/@o3r/training-sdk/package.json"
+        ]
+      }
+    },
+    "publish": {
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "npm publish packages/@o3r/training-sdk/dist"
+      }
+    },
+    "extract-folder-structure": {
+      "cache": true,
+      "executor": "nx:run-script",
+      "options": {
+        "script": "extract"
+      },
+      "inputs": ["source", "^source-and-cli", "{projectRoot}/package.json"],
```

Comment:
Why do you need `^source-and-cli`?
This will require to re-run this action on any dependency source code change.
It is what you want or is `^cli` would be enough (according where `o3r-extract-folder-structure` is)?